coordinator/src/main/scala/filodb.coordinator/queryplanner/MultiPartitionPlanner.scala

```diff
@@ -438,6 +440,51 @@ class MultiPartitionPlanner(partitionLocationProvider: PartitionLocationProvider
     PlanResult(execPlan::Nil)
   }
 
+  def materializeTsCardinalities(lp: TsCardinalities, qContext: QueryContext): PlanResult = {
+
+    def prefixToMatcher(prefix: Seq[String]): String = {
+      val builder = new StringBuilder("{")
+      if (prefix.size > 0) {
+        builder.append(s"""_ws_="${prefix(0)}"""")
+      }
+      if (prefix.size > 1) {
+        builder.append(s""",_ns_="${prefix(1)}"""")
+      }
+      if (prefix.size > 2) {
+        builder.append(s""",__name__="${prefix(2)}"""")
+      }
+      builder.append("}")
+      builder.mkString
+    }
```

Comment:
This bakes in the ws/ns/metric setup...

Maybe the `TsCardinalities` constructor should take a mapping of labels to values?

Reply:
removed